statsd/aggregator.go

```diff
@@ -115,11 +121,14 @@ func (a *aggregator) flushMetrics() []metric {
 	a.counts = countsMap{}
 	a.countsM.Unlock()
 
-	atomic.AddInt32(&a.nbContext, int32(len(counts)))
 	for _, c := range counts {
 		metrics = append(metrics, c.flushUnsafe())
 	}
 
+	atomic.StoreInt32(&a.nbContextCount, int32(len(counts)))
```

Comment:
Can `flushMetrics` be called from different threads?

Reply:
Yes, why ?

Reply:
Is the following case possible?
- THREAD1: run `atomic.StoreInt32(&a.nbContextCount, int32(len(counts)))`
- THREAD2: run `atomic.StoreInt32(&a.nbContextCount, int32(len(counts)))`
- THREAD1: read `a.nbContextCount`. The value returned is the one from THREAD2.

Reply:
Indeed, I moved the `sum` to the telemetry flush